### ISSUE TITLE
git log, git diff failing with Cygwin exception in less

### ISSUE THREAD

**Issue by MichaelJAllen** [git-for-windows/git#1271]
- [x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?

64-bit.

```
$ git --version --build-options

git version 2.14.1.windows.1
built from commit: 82d9b3f3b2407b52251620597d4b14933685459d
sizeof-long: 4
machine: x86_64
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

Windows 10, Version 1607. 64-bit
```
$ cmd.exe /c ver

Microsoft Windows [Version 10.0.14393]
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

```
# One of the following:
> type "C:\Program Files\Git\etc\install-options.txt"
> type "C:\Program Files (x86)\Git\etc\install-options.txt"
> type "%USERPROFILE%\AppData\Local\Programs\Git\etc\install-options.txt"
$ cat /etc/install-options.txt

Path Option: Cmd
SSH Option: OpenSSH
CURL Option: WinSSL
CRLF Option: CRLFAlways
Bash Terminal Option: ConHost
Performance Tweaks FSCache: Enabled
Use Credential Manager: Enabled
Enable Symlinks: Disabled
```

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

In case it's relevant, I do not have Cygwin installed (except for anything from Cygwin installed by git for Windows).

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

Cmd, Git Bash - both have same problem.

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
git log
```
 - What did you expect to occur after running these commands?

Listing of recent commits and log messages.

 - What actually happened instead?

The above command produced the following screen output:
```
      0 [main] less 13356 cygwin_exception::open_stackdumpfile: Dumping stack trace to less.exe.stackdump
```

The file `less.exe.stackdump` has the following content:
```
Stack trace:
Frame        Function    Args
000FFFFBAE0  0018005D31C (000FFFFE3F4, 7FFC4145EC90, 00000000030, 000FFFFDE50)
000FFFFBB80  0018005E91B (00000000064, 00000000000, 00000000000, 00000000000)
000FFFFBDD0  00180121DA0 (000FFFFBF70, 000000017C8, 00000000020, 000FFFFC348)
00180010000  0018011E82E (000007F46A0, 2AE000000000, 00000000000, 00000000288)
000FFFFC2C8  0018011ECB9 (00000000000, 00000000000, 00000000000, 00000000006)
000FFFFC2C8  0018011EE6A (005FCB4228C, 00000000080, 00000000000, 000FFFFC330)
000FFFFC2C8  0018011F11F (005FCB52B06, 0060003A5C0, 005FCB44300, 00600029295)
000FFFFC2C8  00180157346 (00600029295, 000FFFFC2C4, 000FFFFC2C8, 00000000001)
000FFFFC2C8  001800BA3D3 (00000000030, C8CD2DE1433, 0010041A1F8, 000FFFFCC38)
000FFFFC2C8  0018011A58B (00000000030, C8CD2DE1433, 0010041A1F8, 000FFFFCC38)
000FFFFC2C8  005FCB394A4 (000FFFFC6A8, 000FFFFC440, 000007E3AA0, 00000000001)
00000000000  005FCB396F7 (7FFC41450800, 00000000000, 0006828C809, 00000000578)
00000000000  00100401C0B (000FFFFCC30, 0060003A510, 00000000000, 000FFFFCC30)
00000000000  0010041561E (00000000020, 001802D7900, 00180047AD6, 00000000000)
000FFFFCCB0  00180047B47 (00000000000, 00000000000, 00000000000, 00000000000)
00000000000  001800457C3 (00000000000, 00000000000, 00000000000, 00000000000)
End of stack trace (more stack frames may be present)
```

`git diff` produces the same error. I'm guessing that any paged screen output will result in the same error.

 - If the problem was occurring with a specific repository, can you provide the
   URL to that repository to help us with testing?

It fails with every project, but all URL's are local. `git push`, `git clone`, etc. work OK.

**Comment by MichaelJAllen** [comment]
It appears that setting the `TERM` environment variable to `cygwin` solves this problem. The value of `TERM` was `nutc` prior to that.

From looking into that some more, I think _Parametric Creo_ (formerly _Pro/ENGINEER_) is responsible for configuring the environment that way on my machine.

**Comment by tvw** [comment]
The problem persists. It only works whith TERM not being set at all and TERM=cygwin. But is does not work with TERM=vt100 or TERM=ansi. Prior versions worked very well with those settings and setting TERM=cygwin in a non-cygwin-environment is weird and hard to explain to other people who use other programs, which make use of the TERM variable as well.

I use git in 4NT together with a huge bunch of other software.

**Comment by dscho** [comment]
Indeed, there is a regression where anything but the most common `TERM` settings causes problems.

Back when I encountered this issue the first time, I tracked it not all the way down, but just enough to find the work-around. I *think* the problem is somewhere is ncurses, where it no longer handles the absence of `/usr/share/terminfo/$TERM/` (or `/usr/lib/terminfo/$TERM`) gracefully.

@tvw if you have time to spare to track this bug down, I think it would make for an excellent contribution to this project.
